ligand/views.py

```diff
@@ -172,164 +971,145 @@ def CachedTargetDetailsCompact(request, **kwargs):
-                           'protein__species__common_name',
-                           'protein__entry_name',
-                           'protein__name',
-                           'ligand__mw',
-                           'ligand__logp',
-                           'ligand__rotatable_bonds',
-                           'ligand__smiles',
-                           'ligand__hdon',
-                           'ligand__hacc',
-                           'protein',
-                           'publication__web_link__index',
-                           'publication__web_link__web_resource__url',
-                           'affinity',
-                           'potency',
-                           'count_affinity_test',
-                           'count_potency_test',
-                           'reference_ligand'
-                           ).annotate(num_targets=Count('protein__id', distinct=True))
-
-            lig_ids = set([record['ligand__id'] for record in ps])
-            vendor_output = list(LigandVendorLink.objects.filter(ligand_id__in=lig_ids).values_list("ligand_id").annotate(Count('vendor_id', distinct=True)))
-            vendors_dict = {entry[0]:entry[1] for entry in vendor_output}
-
-            for record in ps:
-                record['assay_type'] = record['assay_type'] if record['assay_type'] != 'U' else 'N/A'
-                record['purchasability'] = vendors_dict[record['ligand__id']] if record['ligand__id'] in vendors_dict.keys() else 0
-                record['protein__entry_name'] = record['protein__entry_name'].split('_')[0].upper()
-                record['link'] = record['publication__web_link__web_resource__url'].replace('$index',record['publication__web_link__index']) if record['publication__web_link__web_resource__url'] != None else '#'
-                if record['assay_type'] == 'B':
-                    ligand_data_affinity.append(record)
-                elif record['assay_type'] == 'F':
-                    ligand_data_potency.append(record)
-        context = {}
-        context['affinity_data'] = ligand_data_affinity
-        context['potency_data'] = ligand_data_potency
-        cache.set(cache_key, context, 60*60*24*7)
+        result = {}
+        img_setup_smiles = "https://cactus.nci.nih.gov/chemical/structure/{}/image"
+        d = OrderedDict()
 
-    elif mode == 'compact':
+        for p in ps:
+            if p.ligand not in d:
+                d[p.ligand] = []
+            d[p.ligand].append(p)
 
-        if selection.reference != []:
-            prot_id = [x.item for x in selection.reference]
-            cache_key = "lig_compact_protid_" + ",".join(prot_id)
-            ps = AssayExperiment.objects.filter(protein__in=prot_id).prefetch_related('protein', 'ligand', 'ligand__ligand_type')
-                                                                                      # 'ligand__ids__web_resource',                                                                      # 'ligand')
```

Comment:
ligand and web_resource are not shown or processed in the table anymore?

Reply:
This comes from copy pasting from the Ligands (by receptor query), and yes, web_resource is not shown in the table. 'ligand' is already in the prefetch.

Reply:
Solved in [39bd4a8](https://github.com/protwis/protwis/pull/550/commits/39bd4a8230b189165eddc18d6beb1c8dbe956bd1)